server/tests/stt_parse_content_api_test.py

```diff
@@ -0,0 +1,424 @@
+        self.assertEqual("stt_content_api", self.service.NAME)
+        self.assertEqual("STT Content API", self.service.label)
+        self.assertFalse(self.service.HTTP_AUTH)
+
+        # Check required fields
+        fields = {field["id"]: field for field in self.service.fields}
+        self.assertIn("url", fields)
+        self.assertIn("api_key", fields)
+        self.assertTrue(fields["url"]["required"])
+        self.assertTrue(fields["api_key"]["required"])
+
+    def test_bearer_token_helper(self):
+        """Test the _bearer helper method."""
+        # Test with raw token
+        self.assertEqual("Bearer raw_token", self.service._bearer("raw_token"))
+
+        # Test with Bearer prefix already
+        self.assertEqual(
+            "Bearer existing_token", self.service._bearer("Bearer existing_token")
+        )
+
+    def test_headers_helper(self):
+        """Test the _headers helper method."""
+        headers = self.service._headers("test_api_key")
+
+        self.assertEqual("application/json", headers["Accept"])
+        self.assertEqual("Bearer test_api_key", headers["Authorization"])
+
+    def test_build_params_helper(self):
+        """Test the _build_params helper method."""
+        params = self.service._build_params("2024-01-01T00:00:00Z", 5)
+
+        self.assertEqual({"page": 5}, params)
+
+    def test_config_validation(self):
+        """Test configuration validation in _test method."""
+        # Test missing URL
+        provider = {"config": {"api_key": "test"}}
+
+        with self.assertRaises(Exception):
+            self.service._test(provider)
+
+        # Test missing API key
+        provider = {"config": {"url": "https://example.com"}}
+
+        with self.assertRaises(Exception):
+            self.service._test(provider)
+
+    @responses.activate
+    def test_fetch_data_single_page(self):
+        """Test _fetch_data with single page response using fixture data."""
+        # Use first 2 items from fixture for testing
+        test_items = self.fixture_data["_items"][:2]
+        mock_response_data = {
+            "_items": test_items,
+            "_links": {},  # No next link
+            "_meta": {"total": 2},
+        }
+        url = "https://api.example.com/contentapi/items"
+        responses.add(
+            responses.GET,
+            url,
+            json=mock_response_data,
+            status=200,
+            match=[matchers.query_param_matcher({"page": "1"})],
+        )
+        provider = {
+            "config": {
+                "url": url,
+                "api_key": "Bearer TEST_TOKEN",
+            }
+        }
+        items = self.service._fetch_data(provider, "2024-01-01T00:00:00Z")
+        # Verify a single request was made
+        self.assertEqual(1, len(responses.calls))
+        called_url = responses.calls[0].request.url
+        parsed = urlparse(called_url)
+        qs = parse_qs(parsed.query)
+        self.assertEqual(["1"], qs.get("page"))
```

Comment:
you don't need to parse and check it again, that's already tested via match config in responses.add, it wouldn't match the request if the page=1 wasn't there